docs/adr/0011-operators.md

```diff
@@ -0,0 +1,145 @@
+- Various OLM resources have permission restrictions or behave as a singletons within a namespace.
+- Some operators are not deployed via OLM, yet they behave as operators across all namespaces. Placement of those manifests is also disputed.
+- How should we handle meta-operators which operate across all namespaces and deploy other operators that operate across all namespaces?
+
+## Considered Options
+
+1. Deploy all operator-related resources within `cluster-scope` application.
+2. Require all namespaced resources to be deployed within individual applications, outside of `cluster-scope`.
+3. Find balance between 1. and 2. Deploy resources which have effect on a cluster level within `cluster-scope` app (also the namespace-scoped ones). The rest is deployed in individual applications.
+
+## Decision Outcome
+
+Chosen option: _"3. Find balance between 1. and 2."_, because it is desired to keep track of all resources which may effect other users and other namespaces i.e. operators provisioned to manage all cluster namespaces.
+
+Proposed decision defines strict rules for given resources. These rules are subject to future updates if needed, but the repositories and applications are required to consistently enforce them:
+
+### Operator Lifecycle Manager compatible operators
+
+Operator Lifecycle Manager defines following resource kinds:
+
+- Cluster service version
+- Catalog source
+- Subscription
+- Install plan
+- Operator groups
+
+These are the set of guidelines we want to enforce on them:
+
+#### Cluster service version
+
+This resource is not meant to be deployed manually and is expected to be provided from a _Catalog source_. If there's really a need to deploy a custom _Cluster service version_, the resource has to live within the `cluster-scope` application.
+
+#### Catalog source
+
+It is uncommon for a _Catalog source_ to be declared manually. Operators are expected to be provisioned from the default catalogs and it is advised to rather publish the desired operator to a community catalog than deploying a custom _Catalog source_. If it is really inevitable to deploy an additional `CatalogSource` resource, this resource is required to be deployed to `openshift-marketplace` namespace and therefore it is required to declare this resource within the `cluster-scope` application.
+
+#### Subscription
+
+A `Subscription` resource is a namespace-scoped resource. It is effectively responsible for the operator deployment. Therefore we recognise 2 possible scenarios and use cases for this resource:
+
+1. A namespaced operator with a single namespace visibility:
+
+   The `Subscription` resource can be hosted along with application manifests in an application repository. This application has to be deployed to the same namespace as specified for the subscription.
+
+2. An operator with all namespaces visibility:
+
+   For operators with global effect, we expect the `Subscription` resource to be hosted in `cluster-scope` application only.
```

Comment:
This will need to be included within the `openshift-operators` namespace yes? If so, should we be mentioning that here?

Reply:
Not necessarily. It depends on the operator. A cluster scope operator can be deployed to any namespace if the namespace has an appropriate `OperatorGroup` applied. So for example the cluster logging's elastic search operator is deployed at `openshift-operators-redhat`. It's a mess really, but it can live anywhere.

Reply:
Yeah the elastic search operator is a good point. I suppose my point was that if you use OLM (via console) to install a global operator, the resources go in the `openshift-operator` namespace and it uses the `global-operators`  `operator-group`. 

I was thinking that, for the majority of these cases, we could just be appending the API list for the `global-operator` group, rather than introducing a new namespace for each operator, and adding the OG there. As I understand it that's how it's done via the console but declaratively.

Reply:
The console/UI still allows you to specify the deployment namespace anyways. And sometimes operators have really strong preference on where it should live. I think it's nice to have the possibility but I wouldn't enforce it... things like cluster logging is expected to live in `openshift-logging`, metering operator is expeted to live in `openshift-metering` etc... I mean, there are some OpenShift addons we might need to support and deploy from our cluster-scope app and I would like things to live where they are expected to live. And if it really doesn't matter for the operator, let's have them wherever we see them fit.